src/transcript.rs

```diff
@@ -219,6 +373,50 @@ impl<W: Write, C: CurveAffine> Transcript<C, Challenge255<C>>
     }
 }
 
+impl<W: Write, C: CurveAffine> Transcript<C, Challenge255<C>>
+    for Keccak256Write<W, C, Challenge255<C>>
+{
+    fn squeeze_challenge(&mut self) -> Challenge255<C> {
+        self.state.update(&[KECCAK256_PREFIX_CHALLENGE]);
+
+        let mut state_lo = self.state.clone();
+        let mut state_hi = self.state.clone();
+        state_lo.update(&[KECCAK256_PREFIX_CHALLENGE_LO]);
+        state_hi.update(&[KECCAK256_PREFIX_CHALLENGE_HI]);
+        let hasher_lo = state_lo.clone();
+        let hasher_hi = state_hi.clone();
+        let result_lo: [u8; 32] = hasher_lo.finalize().as_slice().try_into().unwrap();
+        let result_hi: [u8; 32] = hasher_hi.finalize().as_slice().try_into().unwrap();
```

Comment:
Do we need the `hasher_*` variables?

```suggestion
        let result_lo: [u8; 32] = state_lo.finalize().as_slice().try_into().unwrap();
        let result_hi: [u8; 32] = state_hi.finalize().as_slice().try_into().unwrap();
```